static/app/components/route.tsx

```diff
@@ -88,3 +95,41 @@ export function IndexRedirect(_props: IndexRedirectProps) {
   return null;
 }
 IndexRedirect.displayName = 'IndexRedirect';
+
+interface WorkingRedirectProps extends Omit<NavigateProps, 'to'> {
+  /**
+   * Our compat redirect only supports string to
+   */
+  to: string;
+}
+
+/**
+ * Working Redirect component for use in route objects
+ * Wraps a declarative `Navigate` component to interpolate the params of `to`
+ */
+export function WorkingRedirect({to, ...rest}: WorkingRedirectProps) {
```

Comment:
This is a copy of the redirect from the reactrouter6compat module. I'll rename this once we delete the `Redirect` in this file